### PR TITLE
Add Swift API coverage tests for Auth

### PR DESCRIPTION
**Pull request by paulb777**
#no-changelog

### CODE REVIEW

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h [hunk 1]
```diff
@@ -626,7 +626,9 @@ NS_SWIFT_NAME(Auth)
     @param completion Optionally; a block which is invoked when the request finishes. Invoked
         asynchronously on the main thread in the future.
  */
-- (void)checkActionCode:(NSString *)code completion:(FIRCheckActionCodeCallBack)completion;
+- (void)checkActionCode:(NSString *)code
+             completion:
+                 (void (^)(FIRActionCodeInfo *_Nullable info, NSError *_Nullable error))completion;
```
Comment (paulb777) [inline FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h:631]: Improve the Swift non-async await completion.

File: FirebaseAuth/Tests/Sample/AuthSample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -214,13 +216,6 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
-		A00A0EC6CCFF06F1AC46A4F9 /* Pods */ = {
```
Comment (paulb777) [inline FirebaseAuth/Tests/Sample/AuthSample.xcodeproj/project.pbxproj:217]: These get left in file if `pod deprecate` is run when Xcode has the xcworkspace open.

File: FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+import Foundation
+import XCTest
+
+import FirebaseAuth
+import FirebaseCore
+
+/// This file tests public methods and enums. Properties are not included.
+/// Each function maps to a public header file.
+
+class AuthApiBuildOnlyTests: XCTestCase {
```
Comment (ryanwilson) [inline FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift:26]: Nit: prefer `API` to `Api` (https://www.swift.org/documentation/api-design-guidelines/#conventions under "Follow case conventions"). Same with the file name.

File: FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+    _ = AuthErrorCode.unverifiedEmail
+    _ = AuthErrorCode.secondFactorAlreadyEnrolled
+    _ = AuthErrorCode.maximumSecondFactorCountExceeded
+    _ = AuthErrorCode.unsupportedFirstFactor
+    _ = AuthErrorCode.emailChangeNeedsVerification
+    _ = AuthErrorCode.missingOrInvalidNonce
+    _ = AuthErrorCode.missingClientIdentifier
+    _ = AuthErrorCode.keychainError
+    _ = AuthErrorCode.internalError
+    _ = AuthErrorCode.malformedJWT
```
Comment (ryanwilson) [inline FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift:208]: This makes it very obvious to me that we aren't scoping errors well for Swift - ex. `weakPassword` can't be an error on `checkActionCode`. No action needed at the moment but worth evaluating how we can improve this.

File: FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+ */
+
+import Foundation
+import XCTest
+
+import FirebaseAuth
+import FirebaseCore
+
+/// This file tests public methods and enums. Properties are not included.
+/// Each function maps to a public header file.
```
Comment (ryanwilson) [inline FirebaseAuth/Tests/Sample/SwiftApiTests/SwiftApi.swift:24]: I missed this at first and it wasn't immediately obvious to me there, what do you think about appending `_h` to the name of the functions?

**Review by ryanwilson — APPROVED** [review]
Overall LGTM

**Review by ryanwilson — APPROVED** [review]

**Pull request merged**